Feedback on Getting Started Vignette (#18)
FredHutch/gimap: vignettes/getting-started.Rmd

gimap performs analysis of dual-targeting CRISPR screening data, with the goal of aiding the identification of genetic interactions (e.g. cooperativity, synthetic lethality) in models of disease and other biological contexts. gimap analyzes functional genomic data generated by the pgPEN (paired guide RNAs for genetic interaction mapping) approach, quantifying growth effects of single and paired gene knockouts upon application of a CRISPR library. A multitude of CRISPR screen types can be used for this analysis, with helpful descriptions found in this review (https://www.nature.com/articles/s43586-021-00093-4). Use of pgPEN and GI-mapping in a paired gRNA format can be found here (https://pubmed.ncbi.nlm.nih.gov/34469736/). 

```{r}

> library(dplyr)

Author: If we are going to use dplyr, we might as well load dplyr, which loads magrittr.

Reviewer: Sounds good to me! Can you add this in too?

library(gimap)
```

## Data requirements 

Let's examine this example pgPEN counts table. It's divided into columns containing: 

- `id`: an ID corresponding to the names of paired guides
- `seq_1`: gRNA sequence 1, targeting "paralog A"
- `seq_2`: gRNA sequence 2, targeting "paralog B"

> - The sample, day, and replicate number for which gRNAs were sequenced (TODO)

Author: I was unclear what the last bullet point was saying. The column names in `example_data` that this point is referencing, `Day00_RepA`, `Day05_RepA`, `Day22_RepA` `Day22_RepB` `Day22_RepC`, doesn't seem to have any info on the sample. I think  the names refer to the Day, and Replicate number, but not the sample. 

I would list out each column and describe what each column is. For ex,

- `Day00_RepA`: Gene Counts from Day 00 for Replicate A 
- `Day05_RepA`: Gene Counts from Day 05 for Replicate A 
- `Day22_RepA`: Gene Counts from Day 22 for Replicate A 
- `Day22_RepB`: Gene Counts from Day 22 for Replicate B

Reviewer: This also sounds great! Can you add this?

```{r}
example_counts <- example_data %>%

>   select(c("Day00_RepA", "Day05_RepA", "Day22_RepA", "Day22_RepB", "Day22_RepC")) %>%

Author: Lets get rid of the `pkg::fun` syntax as it makes code hard to read.

Reviewer: This is a matter of opinion. I find it easier to read code this way because I know which function we are calling (some packages have functions of the same name). 

I also don't like loading a full package if we only are using one function from it. 

However that being said I'm not opposed to suggestion -- I think in this context using `library(dplyr)` might be the right move for this audience. The main thing is to be consistent.

Reviewer: So yes, I agree!

You'll notice that this set up gives us a list of formatted data. This contains the original counts we gave `setup_data()` function but also normalized counts, and the total counts per sample. 

> 

Author: `gimap_dataset` is a list with 5 elements inside. Let's list out these elements and give a 1-sentence description:

For ex,

- `raw_counts`: Raw counts
- `counts_per_sample`: Counts per sample
- `coverage`: 
- `transformed_data`: Transformed dataset
- `metadata`: Metadata

Reviewer: I like this! Can you add this in @howardbaek ?

Author: Can @danieljgroso or anyone else give more details on these elements?

Reviewer: I think we ended up covering this in the meeting but let me know if there's any info you need.

You'll notice that this set up gives us a list of formatted data. This contains the original counts we gave `setup_data()` function but also normalized counts, and the total counts per sample. 

> - Raw counts: Original data

Reviewer: ```suggestion
- `raw_counts`: The original counts data that illustrates the number of cells alive in the sample. This data has samples as the columns and the paired guide constructs as rows. 
```

This kinda formatting helps.

> - Counts per Sample: Add up all the counts for each sample

Reviewer: ```suggestion
- `counts_per_sample`: Add up all the counts for each sample over all of the paired guide designs. 
```

> - Transformed data: Contains normalized counts, counts per million (cpm)

Reviewer: ```suggestion
- Transformed data: This section contains the various types of normalized and adjusted data made from the raw counts data. 
```

> - Log2 CPM: log-2 transformed CPM

Reviewer: ```suggestion
- `count_norm` - For each sample, the data is normalized `-log10(( counts +1) /  total counts for the sample over all the pg designs ))`
- `cpm` - For each sample this is calculated by taking the `counts / total counts for the sample over all the pg designs)*1e6`
- `log2cpm`: log-2 transformed counts per million this is calculated by `log2(cpms + 1)`
```

> - pg_metadata: paired guide metadata

Reviewer: ```suggestion
- pg_metadata: paired guide metadata - information that describes the paired-guided RNA designs. This may include the sequences used in the CRISPR design as well as what genes are targeted. 
```

> - sample_metadata: Metadata of dataset where columns are different samples and rows are different paired guide.

Reviewer: ```suggestion
- `sample_metadata`: Metadata that describes the samples. This likely includes the time point information, replicates, sample IDs, and any other additional information that is needed regarding the experimental setup. 
```

> 

Reviewer: Probably can delete these extra lines.
